hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/validator/SqlQueryEqualityPreCommitValidator.java

```diff
@@ -54,13 +54,11 @@ protected String getQueryConfigName() {
 
   @Override
   protected void validateUsingQuery(String query, String prevTableSnapshot, String newTableSnapshot, SQLContext sqlContext) {
-    String queryWithPrevSnapshot = query.replaceAll(HoodiePreCommitValidatorConfig.VALIDATOR_TABLE_VARIABLE, prevTableSnapshot);
-    String queryWithNewSnapshot = query.replaceAll(HoodiePreCommitValidatorConfig.VALIDATOR_TABLE_VARIABLE, newTableSnapshot);
-    LOG.info("Running query on previous state: " + queryWithPrevSnapshot);
-    Dataset<Row> prevRows = sqlContext.sql(queryWithPrevSnapshot).cache();
+    LOG.info("Running query on previous state: " + prevTableSnapshot);
+    Dataset<Row> prevRows = executeSqlQuery(sqlContext, query, prevTableSnapshot).cache();
```

Comment:
It looks like a simple refactoring. Trying to understand the root cause. Was it because of successive `replaceAll` calls before executing prev snapshot and new snapshot queries?

Reply:
Sorry, the actual fix is in `SparkValidatorUtils` which was somehow revert in my force push.  Fixed now.